src/injected/shadow-utils.ts

```diff
@@ -7,9 +7,9 @@ export class ShadowUtils {
     constructor(htmlElementUtils: HTMLElementUtils) {
         this.htmlElementUtils = htmlElementUtils;
     }
-    public getShadowContainer(): HTMLElement {
+    public getShadowContainer(): HTMLElement | undefined {
         const shadowHost = this.htmlElementUtils.querySelector('#insights-shadow-host');
 
-        return shadowHost.shadowRoot.querySelector('#insights-shadow-container') as HTMLElement;
+        return shadowHost?.shadowRoot?.querySelector('#insights-shadow-container') as HTMLElement;
```

Comment:
Every callsite where `getShadowContainer()` is called assumes that it will be returned as defined; based on that, I'd recommend having this throw an error rather than return undefined, since it doesn't introduce any new error paths and would improve the message in existing error paths (`cannot invoke method appendChild on undefined` turns into `Could not find shadow container element #insights-shadow-host`)